public/components/incontext_insight/index.tsx

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './index.scss';
+
+import { i18n } from '@osd/i18n';
+import {
+  EuiWrappingPopover,
+  EuiButton,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiPopoverTitle,
+  EuiText,
+  EuiPopoverFooter,
+  EuiBadge,
+  EuiSpacer,
+  EuiListGroup,
+  EuiListGroupItem,
+  EuiPanel,
+  keys,
+  EuiIcon,
+  EuiButtonIcon,
+} from '@elastic/eui';
+import React, { Children, isValidElement, useEffect, useRef, useState } from 'react';
+import { IncontextInsight as IncontextInsightInput } from '../../types';
+import { getIncontextInsightRegistry, getChrome, getNotifications } from '../../services';
+
+export interface IncontextInsightProps {
+  children: React.ReactNode;
+}
+
+// TODO: add saved objects / config && i18n
+const container = document.createElement('div');
+container.id = 'incontext-insight-target';
+container.setAttribute('data-enabled', 'false');
+document.body.appendChild(container);
+
+export const IncontextInsight = ({ children }: IncontextInsightProps) => {
+  const anchor = useRef<HTMLDivElement>(null);
+  const [isVisible, setIsVisible] = useState(false);
+
+  useEffect(() => {
+    // TODO: use animation when not using display: none
+    if (anchor.current) {
+      const incontextInsightAnchorButtonClassList = anchor.current.parentElement?.querySelector(
+        '.incontextInsightAnchorButton'
+      )?.classList;
+      const incontextInsightAnchorIconClassList = anchor.current.querySelector(
+        '.incontextInsightAnchorIcon'
+      )?.children[0].classList;
+
+      if (!incontextInsightAnchorButtonClassList || !incontextInsightAnchorIconClassList) return;
+
+      incontextInsightAnchorButtonClassList.add('hoverEffectUnderline');
+      incontextInsightAnchorIconClassList.add(
+        'hoverEffect25',
+        'hoverEffect50',
+        'hoverEffect75',
+        'hoverEffect100'
+      );
+
+      setTimeout(() => {
+        incontextInsightAnchorIconClassList.remove('hoverEffect100');
+        setTimeout(() => {
+          incontextInsightAnchorIconClassList.remove('hoverEffect75');
+          setTimeout(() => {
+            incontextInsightAnchorIconClassList.remove('hoverEffect50');
+            setTimeout(() => {
+              incontextInsightAnchorIconClassList.remove('hoverEffect25');
+              incontextInsightAnchorButtonClassList.remove('hoverEffectUnderline');
+            }, 10);
+          }, 15);
+        }, 20);
+      }, 1125);
```

Comment:
Can we make it a css animation?

Reply:
I want to this code feels hacky and im willing to remove. this is basically only for the onload state and to avoid having the icon on hover push other elements while also be adding itself to the anchor and not causing issues to anchor as well before displaying block.

this is the biggest factor to decide whether we should allow this icon to show on any element

Reply:
i have cleaned up the code to make it a little bit cleaner